src/main/java/com/microsoft/jenkins/keyvault/BaseSecretCredentials.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ */
+
+package com.microsoft.jenkins.keyvault;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import com.google.common.annotations.VisibleForTesting;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+import com.microsoft.azure.util.AzureCredentials;
+import hudson.model.Item;
+import hudson.security.ACL;
+import hudson.util.ListBoxModel;
+
+public class BaseSecretCredentials extends BaseStandardCredentials {
```

Comment:
Actually it's already supported because I implement them as standard string/certificate credentials. No new credential type is added. Users can bind KeyVault value as environment variable by choosing `Secret Text` in Credentials Bindings Plugin.